src/Service/DashboardService.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\User;
+use App\Entity\Worklog;
+use App\Model\DashboardData;
+use App\Repository\WorkerRepository;
+use App\Repository\WorklogRepository;
+
+class DashboardService
+{
+    public function __construct(
+        private readonly WorkerRepository $workerRepository,
+        private readonly WorklogRepository $worklogRepository,
+    ) {
+    }
+
+    public function getUserDashboard(User $user, ?int $year = null): ?DashboardData
+    {
+        $userEmail = $user->getEmail();
+
+        if (null === $userEmail) {
+            return null;
+        }
+
+        $worker = $this->workerRepository->findOneBy(['email' => $userEmail]);
+
+        if (null === $worker) {
+            return null;
+        }
+
+        $norm = $worker->getWorkload() ?? 0.0;
+
+        if (null == $year) {
+            $year = (int) date('Y');
+        }
+
+        $yearStart = new \DateTime();
+        $yearStart->setDate($year, 1, 1);
+        $yearStart->setTime(0, 0);
+
+        $yearEnd = new \DateTime();
+        $yearEnd->setDate($year + 1, 1, 1);
+        $yearEnd->setTime(0, 0);
```

Comment:
DateTimeHelper::getFirstAndLastDateOfYear can provide this if you want to streamline a bit